app/Http/Controllers/SaveController.php

```diff
@@ -155,16 +187,27 @@ public function update(Request $request, Save $save): Response
                 "data" => "nullable|json",
                 "name" => "string|max:255",
                 "description" => "string|max:300",
+                "resources.*" => [
+                    File::types(self::ALLOWED_MIMETYPES)
+                        ->max(self::FILE_MAX_KILOBYTES)
+                ],
                 "lock" => "prohibited"
             ]);
 
-            if ($save->locked_by_id === $user->id) {
-                $save->fill($validated);
-                $save->save();
-                return response()->noContent(Response::HTTP_OK);
-            } else {
-                return response()->noContent(Response::HTTP_LOCKED);
-            }
+            return DB::transaction(function () use ($validated, $save, $user) {
+                if ($save->locked_by_id === $user->id) {
+                    $save->fill($validated);
+                    $save->save();
+
+                    if (array_key_exists("resources", $validated)) {
+                        $this->resourceService->saveResources($save, $validated["resources"]);
+                    }
```

Comment:
in the new commit i made it so that if no resources are in the request, the resources will be deleted. This can be done because the blobs of the resources are now synced with the frontend in the ResourceManager.